sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1753,6 +1753,27 @@ class Dataset[T] private[sql](
    *
    * Also as standard in SQL, this function resolves columns by position (not by name).
    *
+   * Notice that the column positions in the schema aren't necessarily matched with the
+   * fields in the typed objects in a Dataset. This function resolves columns by their positions
+   * in the schema, not the fields in the typed objects:
+   *
+   * {{{
```

Comment:
Could we clarify this is a Scala example?

Reply:
Sure.